src/pages/Popup/scripts/alarm.js

```diff
@@ -116,18 +120,38 @@ async function AddAlarm_click(course, dates) {
 async function AddCustomAlarm(alarm) {
   var resp = await new Promise((resolve) => chrome.storage.sync.get(resolve));
   var data = resp['Defaults'];
+  var authuser = data['Authuser'];
   var beforeminutes = data['BeforeMinutes'];
   var beforeseconds = data['BeforeSeconds'];
+  var autojoin = data['AutoJoin'];
+  var autoleaveswitch = data['AutoLeaveSwitch'];
+  var endtimecheck = data['EndTimeCheck'];
   var course = alarm;
   course['type'] = 'custom';
   var start_date = new Date(alarm.Time);
+  if (alarm.EndTime != '' && autoleaveswitch == true) {
+    endtimecheck = true;
+  }
+  var values = {
+    Authuser: authuser,
+    BeforeMinutes: beforeminutes,
+    BeforeSeconds: beforeseconds,
+    AutoJoin: autojoin,
+    AutoLeaveSwitch: autoleaveswitch,
+    EndTimeCheck: endtimecheck
```

Comment:
I realized that setting `EndTimeCheck` in storage and using the stored value whenever an alarm is triggered can lead to a big issue. 
Say I set an alarm (with no end time) for today at 8 pm. So in this case, `EndTimeCheck` won't be true and the old value would be stored in storage (the old value can be true or false)
Now, I set a new alarm (with end time) for today at 9 pm. In this case, `EndTimeCheck` will be true and we are saving that in storage.
When the 8 pm alarm gets triggered, we use the default values from storage. According to the values, `EndTimeCheck` would be true, but  in this case, we did not set the end time value, so it should have been false

So what we should do is, check `alarm.EndTime != ''` and  `autoleaveswitch == true` in background.js when the alarm is triggered. If it satisfies both the condition, we pass true, else false.

Reply:
Hmm makes sense. I'll change it.